File: code_samples/lua/security_input_validation.lua
-- Security: Input Validation
-- AI/ML Training Sample

InputValidation = {}
InputValidation.__ index = InputValidation

function InputValidation:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function InputValidation:process(input)
    self.data = input
end

function InputValidation:getData()
    return self.data
end

function InputValidation:validate()
    return #self.data > 0
end

-- Example usage
local instance = InputValidation:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))

File: code_samples/go/security_input_validation.go
package main

import (
    "fmt"
)

// Security: Input Validation
// AI/ML Training Sample

type InputValidation struct {
    Data string
}

func NewInputValidation() *InputValidation {
    return &InputValidation{
        Data: "",
    }
}

func (s *InputValidation) Process(input string) {
    s.Data = input
}

func (s *InputValidation) Validate() bool {
    return len(s.Data) > 0
}

func (s *InputValidation) GetData() string {
    return s.Data
}

func main() {
    instance := NewInputValidation()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
